### PR TITLE
add stage-out API

### PR DESCRIPTION
**Pull request by mlassnig**

### CODE REVIEW

**Comment by mlassnig** [comment]
hmmm, travis is not happy with creating directories.... need to check why, works locally

File: pilot/api/data.py [hunk 1]
```diff
@@ -47,6 +49,51 @@ def transfer(self, files):
             raise Exception('Files dictionary does not conform: scope, name, destination')
 
 
+class StageOutClient(object):
+
+    def __init__(self, site=None):
+        super(StageOutClient, self).__init__()
+
+        # Check validity of specified site
+        self.site = os.environ.get('VO_ATLAS_AGIS_SITE', site)
```
Comment (PalNilsson) [inline pilot/api/data.py]: Doesn't look good with an ATLAS hardcoding - is anyone ever setting VO_ATLAS_AGIS_SITE? I don't see it in any logs. Do we need it?

File: pilot/api/data.py [hunk 2]
```diff
@@ -47,6 +49,51 @@ def transfer(self, files):
...
+        Automatically stage out files using rucio.
+
+        :param files: List of dictionaries containing the target scope, the path to the file, and destination RSE.
+                      [{scope, file, rse
+                      Additional variables that can be used:
+                        lifetime                        of the file in seconds
+                        no_register                     setting this to True will not register the file to Rucio (CAREFUL!)
+                        guid                            manually set guid of file in either string format
+                                                         XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX or
+                                                         XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
```
Comment (PalNilsson) [inline pilot/api/data.py]: The formats are the same..?

Comment (mlassnig) [inline pilot/api/data.py]: whoops, typo

File: pilot/api/data.py (hunk 1 above)
Comment (mlassnig) [inline pilot/api/data.py]: this is set by the ALRB...

File: pilot/control/data.py [hunk 3]
```diff
@@ -165,6 +165,84 @@ def stage_in_auto(site, files):
...
+    """
+
+    # don't spoil the output, we depend on stderr parsing
+    os.environ['RUCIO_LOGGING_FORMAT'] = '%(asctime)s %(levelname)s [%(message)s]'
+
+    executable = ['/usr/bin/env',
+                  'rucio', '-v', 'upload']
+
+    # quickly remove non-existing destinations
+    for file in files:
```
Comment (PalNilsson) [inline pilot/control/data.py]: 'file' is a builtin - can you change it to _file or something instead?

File: pilot/api/data.py (hunk 1 above)
Comment (PalNilsson) [inline pilot/api/data.py]: But do we care..? You could anyway set it like
data_client = data.StageOutClient(site='ALCF_Theta')
right?

**Comment by PalNilsson** [comment]
Ok fine, thanks!

**Pull request merged**